test/common/core/corediff.scenarios.spec.js

```diff
@@ -575,6 +575,185 @@ describe('corediff scenarios', function () {
             .nodeify(done);
     });
 
+    it('should give conflict when container is removed while other branch added children', function (done) {
+        var originRoot,
+            basePath,
+            containerPath;
+
+        loadRootAndFCO(context.rootHash)
+            .then(function (r) {
+                var base = core.createNode({
+                        parent: r.root,
+                        base: r.fco
+                    }),
+                    container = core.createNode({
+                        parent: r.root,
+                        base: r.fco
+                    });
+
+                basePath = core.getPath(base);
+                containerPath = core.getPath(container);
+
+                originRoot = r.root;
+                return save(r.root);
+            })
+            .then(function (rootHash) {
+                return Q.all([
+                    loadRootAndFCO(rootHash, [basePath, containerPath]),
+                    loadRootAndFCO(rootHash, [basePath, containerPath]),
+                ]);
+            })
+            .then(function (trees) {
+                // We've loaded two trees from the same rootHash
+                // now let's make changes.
+
+                core.createNode({
+                    parent: trees[0][containerPath],
+                    base: trees[0].fco,
+                    relid: 'childOne'
+                });
+                core.createNode({
+                    parent: trees[0][containerPath],
+                    base: trees[0].fco,
+                    relid: 'childTwo'
+                });
+
+                core.deleteNode(trees[1][containerPath]);
+
+                // Save to ensure the added nodes are persisted.
+                return Q.all([
+                    save(trees[0].root),
+                    save(trees[1].root)
+                ])
+                    .then(function () {
+                        return Q.all([
+                            core.generateTreeDiff(originRoot, trees[0].root),
+                            core.generateTreeDiff(originRoot, trees[1].root),
+                        ]);
+                    });
+            })
+            .then(function (diffs) {
+                var concatChanges01 = core.tryToConcatChanges(diffs[0], diffs[1]),
+                    concatChanges10 = core.tryToConcatChanges(diffs[1], diffs[0]);
```

Comment:
Are you sure the diffs aren't mutated as part of the `tryToConcatChanges`?

Reply:
I am positive. Double checked it. We can add it though to the test if you wish.